backend/src/routes/api/storage/index.ts

```diff
@@ -0,0 +1,47 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { createMinioClient, getObjectStream } from './storageUtils';
+
+export default async (fastify: FastifyInstance): Promise<void> => {
+  fastify.get('/:namespace/:bucket', async (request: FastifyRequest, reply: FastifyReply) => {
+    try {
+      const { namespace, bucket } = request.params as {
+        namespace: string;
+        bucket: string;
+        key: string;
```

Comment:
Key shouldn't be here, right? key is in the query...

Reply:
fixed